File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        if (nums.length ==0) return 0;
        let num = nums.sort((a,b)=> a-b);
        let count = 1;
        let longest = 1;
        for (let i = 0; i<num.length;i++){
            if (num[i] == num[i+1]) continue;
            
            if(num[i]+1 == num[i+1]){
                count = count+1;
            }else if( num[i]+1 != num[i+1]){
                count = 1;
            }
            longest = Math.max(count,longest);
        }
        return longest;
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-4.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        res = []

        for i,n in enumerate(nums):
            if (i > 0):
                if n == nums[i-1]:
                    continue
            target = -n
            # target = x + y
            l, r = i+1, len(nums)-1
            while (l < r):
                if nums[l] + nums[r] < target:
                    l+=1
                elif nums[l] + nums[r] > target:
                    r-=1
                else:
                    res.append([nums[l],nums[r],n])
                    l+=1
                    while l<r and nums[l] == nums[l-1]:
                        l+=1
                    r-=1
                    while l<r and nums[r] == nums[r+1]:
                        r-=1
        
        return res
                
